tedana/decomposition/ica.py

```diff
@@ -50,6 +53,51 @@ def tedica(data, n_components, fixed_seed, maxit=500, maxrestart=10):
         "decompose the dimensionally reduced dataset."
     )
 
+    if ica_method=='robustica':
+        mmix, Iq = r_ica(data, n_components, n_robust_runs, maxit)
+        fixed_seed=-99999
+    elif ica_method=='fastica':
+        mmix, fixed_seed=f_ica(data, n_components, fixed_seed, maxit=500, maxrestart=10)
+        Iq = 0
+    else:
+        LGR.warning("The selected ICA method is invalid!")
+        sys.exit()
```

Comment:
1. A "warning", to me, is appropriate in scenarios where there is potentially some issue, or some ambiguity where the software is being forced to make a decision it doesn't want to have to make, but it can nevertheless proceed. This is the opposite of the scenario where the next line of code results in termination of execution.

2.  Are there usage scenarios where this function is executed directly without having gone through the command-line interface? If *not*, then given that the command-line interface explicitly restricts the set of choices that can be made, it should be impossible for code execution to reach this point. In Python I tend to use an `assert False` so that the interpreter / linter understand this.

3.  `sys.exit()` should not be used unless absolutely necessary. Check elsewhere in the tedana code to see how they deal with premature termination of execution; raising of an appropriate Exception would be preferable.

4.  `sys.exit()` should not be called without specifying an exit code. In this case, execution was not successful, and so the exit code should be non-zero.

Reply:
"sys.exit" is replace with raise ValueError. This is what they have done throughout the code.